pkg/controller/storagecluster/cephcluster.go

```diff
@@ -475,3 +484,29 @@ func newCephDaemonResources(custom map[string]corev1.ResourceRequirements) map[s
 
 	return resources
 }
+
+func (r *ReconcileStorageCluster) throttleStorageDevices(storageClassName string) (bool, error) {
+	storageClass := &storagev1.StorageClass{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: "", Name: storageClassName}, storageClass)
+	if err != nil {
+		return false, fmt.Errorf("failed to retrieve StorageClass %q. %+v", storageClassName, err)
+	}
+	bool flag := false
+	switch storageClass.Provisioner {
+	case string(EBS):
+		if contains(throttleDiskTypes, storageClass.Parameters["type"]) {
+			flag = true
+			sc.Spec.StorageDeviceSets[i].Config.TuneDeviceClass = true
+		}
+	}
+	case string(Premium):
+		if contains(throttleFastDiskTypes, storageClass.Parameters["type"]) {
+			flag = true
+			sc.Spec.StorageDeviceSets[i].Config.TuneFastDeviceClass = true
+		}
+	if flag {
+		sc.Spec.StorageDeviceSets[i].Config.TuneFastDeviceClass = false
+		sc.Spec.StorageDeviceSets[i].Config.TuneDeviceClass = false
```

Comment:
You need to update SC at the end before return, So these changes will be saved.

Reply:
```suggestion
		sc.Spec.StorageDeviceSets[i].Config.TuneFastDeviceClass = false
```

Reply:
> You need to update SC at the end before return, So these changes will be saved.

And would be better if done in ensureCephCluster as updating it in this function could mean possibly updating the storageCluster for each storageDeviceSet.